pkg/trace/api/api.go

```diff
@@ -596,6 +603,17 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 	}
 }
 
+const outOfCPULogThreshold uint32 = 5
+
+// logOutOfCPU logs a warning on the first call and every n+outOfCPULogThreshold calls.
+// It should be called when the receiver spawns a new Goroutine to handle new requests.
+func (r *HTTPReceiver) logOutOfCPU() {
+	count := r.outOfCPUCounter.Inc()
```

Comment:
IMHO this doesn't need a separate method. But I'll let you decide if you prefer it.